### PR TITLE
Add UDP output support

### PR DESCRIPTION
**Pull request by andreapignaz**
Implementation of basic UDP output supporting the [spiffer protocol](https://github.com/SpiNNakerManchester/spif/tree/master/spiffer). CLI support added. Unit test added. Timestamps aren't supported at the moment. Example of usage: `faery input stdin output udp localhost 12000`.

### CODE REVIEW

**Comment by andreapignaz** [comment]
Timestamp support have been added in the last commit, following the spiffer protocol. Unit Test was updated accordingly.

**Comment by andreapignaz** [comment]
The buffer in which the packets are stored before being sent via UDP has been optimized using `numpy` and a preallocated array. 
The `encode_spif` function is using standard python types and not numpy types because it seems that converting to `np.uint32` takes longer than converting to `int` after some tests on my machine.

**Review by Jegp — APPROVED** [review]
Very nicely done. I think this is exactly the functionality we'd need in the first pass. Also nice with the added timestamp feature :)
I'm happy to merge it as-is. I do have one remark about the way you're or'ing the bits together for the UDP message. It looks correct, but I think we improve the code by removing a redundant operation.

File: faery/udp.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        self.include_ts = include_timestamps
+        # Buffer = 512 bytes / 4 bytes per event. Timestamps, if sent, are other 4 bytes, and another element in the buffer.
+        self.max_buffer_size = 512/4 
+        self.buffer: np.ndarray = np.empty(int(self.max_buffer_size), dtype=np.uint32)
+        self.buffer_idx: int = 0
+        self.kwargs = kwargs
+
+    def encode_spif(self, event: Event) -> int:
+        message = int(event['y'])
+        message |= message | (int(event['p']) << 15)
```
[the remark this answers is not in the document]
Comment (andreapignaz) [reply faery/udp.py]: You're totally right. I solved this in the next commit.

File: faery/udp.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            self.buffer_idx += 1
+            if self.include_ts:
+                timestamp = event['t']
+                self.buffer[self.buffer_idx] = timestamp
+                self.buffer_idx += 1
+            if self.buffer_idx >= self.max_buffer_size:
+                self.clientSocket.sendto(self.buffer.tobytes(), self.addr)
+                self.buffer_idx = 0
+        if self.buffer_idx > 0:
+            self.clientSocket.sendto(self.buffer[0:self.buffer_idx].tobytes(), self.addr)
```
[the remark this answers is not in the document]
Comment (andreapignaz) [reply faery/udp.py:42]: I think that next steps for this udp implementation is understanding how fast it is, and finding proper solutions. For example, I think we need to understand if this solution is capable of streaming real camera data in real time. 
If we go for parallelization, we need to make sure that the packet order is respected.

[the remark this answers is not in the document]
Comment (Jegp) [reply faery/udp.py:42]: Agreed. As we know premature optimization [is the root of all evil](https://www.laws-of-software.com/laws/knuth/) :smiling_imp:
I do wonder how important the packet order is, but we can pick that up later.

**Pull request merged**